RadarSDK/Radar.m

```diff
@@ -383,80 +391,55 @@ + (void)rejectEventId:(NSString *)eventId {
+                       completionHandler:(RadarLogConversionCompletionHandler) completionHandler {
+    [[RadarAPIClient sharedInstance] sendEvent:name withMetadata:metadata completionHandler:^(RadarStatus status, NSDictionary * _Nullable res, RadarEvent * _Nullable event) {
         if (status != RadarStatusSuccess) {
             if (completionHandler) {
                 [RadarUtils runOnMainThread:^{
-                    completionHandler(status, nil, nil, nil);
+                    completionHandler(status, nil);
                 }];
             }
 
             return;
         }
+        
+        if (completionHandler) {
+            [RadarUtils runOnMainThread:^{
+                completionHandler(status, event);
+            }];
+        }
+    }];
+}
 
-        [[RadarAPIClient sharedInstance] sendEvent:customType
-                                      withMetadata:metadata
-                                              user:user
-                                       trackEvents:events
-                                 completionHandler:^(RadarStatus status, NSDictionary *_Nullable res, NSArray<RadarEvent *> *_Nullable events) {
-                                     if (status != RadarStatusSuccess) {
-                                         if (completionHandler) {
-                                             [RadarUtils runOnMainThread:^{
-                                                 completionHandler(status, nil, nil, nil);
-                                             }];
-                                         }
-
-                                         return;
-                                     }
++ (void)logConversionWithName:(NSString *)name
+                     metadata:(NSDictionary *_Nullable)metadata
+            completionHandler:(RadarLogConversionCompletionHandler)completionHandler {
+    NSTimeInterval lastTrackedTimeInterval = [[NSDate date] timeIntervalSinceDate:[RadarSettings lastTrackedTime]];
+    BOOL isLastTrackRecent = lastTrackedTimeInterval < 60;
 
-                                     if (completionHandler) {
-                                         [RadarUtils runOnMainThread:^{
-                                             completionHandler(status, location, events, user);
-                                         }];
-                                     }
-                                 }];
+    CLAuthorizationStatus authorizationStatus = [[RadarLocationManager sharedInstance].permissionsHelper locationAuthorizationStatus];
+    if (authorizationStatus == kCLAuthorizationStatusDenied || isLastTrackRecent) {
+        [self sendLogConversionRequestWithName:name metadata:metadata completionHandler:completionHandler];
+        
+        return;
+    }
+    
+    [self trackOnceWithCompletionHandler:^(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarEvent *> * _Nullable events, RadarUser * _Nullable user) {
```

Comment:
@joeychik-radar @david-goodfellow @jsani-radar 

Why are we doing a `trackOnceWithCompletionHandler` here if we don't pass in any of the response data from it to `sendLogConversionRequestWithName`?  Related, why are we checking for a recent track request?

Reply:
the standard customer flow is `trackOnce` --> surface in-app use case --> log conversion event - if logging the conversion event were to call `trackOnce` again, that'd be redundant

so, there's a check if a recent enough location fix has been grabbed - if not, then we should flush the user state with a new one to attribute the right location context to the conversion

Reply:
"to attribute the right location context to the conversion" - Are we attaching a lat long to the Event here or does this all happen on the server side?

Reply:
all the location context (geofence, beacon, trip, etc) is attached server side